### PR TITLE
Restore stage separated Jenkinsfile

### PR DESCRIPTION
**Pull request by joakime**
Restore the Stage separated Jenkinsfile from before.

This has the side effect that our Jenkins/CI builds **does not follow the same rules as a maven command line build** (or IDE build).
Building locally will **not** produce the same results as Jenkins/CI.
It is extremely likely that CI will fail, but local build will not.
If you encounter that, blame this stage separated `Jenkinsfile`.

### CODE REVIEW

**Comment by gregw** [comment]
@joakime nice to have the staged view back in CI.  But the main thing I was asking for was the restoration of fail at end semantics.  Currently CI is stopping at the first test failure, which may be a flake, and thus we don't get to see the rest of the build.    

So does this PR restore fail-at-end behaviour as well?

**Comment by olamy** [comment]
@gregw fail at end has been restored in jetty-12.0.x branch with this commit https://github.com/eclipse/jetty.project/commit/f5e3ba459a77ac9ea8ffe2cf743b666be8dcec61 
which means all tests are executed even if a module has a test failure

**Comment by olamy** [comment]
@joakime LGTM maybe we could add `-Dmaven.test.failure.ignore=true` as well to run all tests?

**Comment by joakime** [comment]
> @joakime LGTM maybe we could add `-Dmaven.test.failure.ignore=true` as well to run all tests?

Sure, but lets put it in the `mavenBuild` function instead of in every stage.

File: Jenkinsfile [hunk 1]
```diff
@@ -103,7 +103,7 @@ pipeline {
               timeout(time: 120, unit: 'MINUTES') {
                 dir("${env.WORKSPACE}/buildy") {
                   //cleanup all projects
-                  mavenBuild("jdk17", "clean install -f jetty-home", "maven3")
+                  mavenBuild("jdk17", "clean install -pl :jetty-home", "maven3")
```
Comment (joakime) [inline Jenkinsfile:106]: Why this change?
